indexer.py

```diff
@@ -183,27 +200,27 @@ async def process_file(
 
     logging.debug(f"{filepath} gathered.")
     logging.debug(metadata)
-    await request_post_patch(fc_rc, metadata, no_patch, dryrun)
+    await post_metadata(fc_rc, metadata, patch, dryrun)
 
 
-async def process_paths(
+async def index_paths(
     paths: List[str],
     manager: MetadataManager,
     fc_rc: RestClient,
-    no_patch: bool,
+    patch: bool,
     dryrun: bool,
 ) -> List[str]:
-    """POST metadata of files given by paths, and return any directories."""
-    sub_files: List[str] = []
+    """POST metadata of files given by paths, and return all child paths."""
+    child_paths: List[str] = []
 
-    for p in paths:
+    for p in paths:  # pylint: disable=C0103
         try:
             if is_processable_path(p):
                 if os.path.isfile(p):
-                    await process_file(p, manager, fc_rc, no_patch, dryrun)
+                    await index_file(p, manager, fc_rc, patch, dryrun)
```

Comment:
If you want even more performance, you can run multiple indexers in parallel, which may help if it is waiting on a FC response.  Example: https://github.com/WIPACrepo/iceprod/blob/master/iceprod/server/scheduled_tasks/update_task_priority.py#L66

Reply:
I'll keep that in mind if it becomes a noticeable issue. The FC turnaround-time actually isn't all that bad usually. The checksums and I3Reader reading are the most expensive ops by far.

Reply:
What would be nice is to re-design the multi-processing because currently the parent process isn't doing much, which just lowers CPU utilization.